### PR TITLE
[type:feat]Alert module adapts to namespace transformation

### PR DESCRIPTION
**Pull request by 478320**
Make sure that:

- [x] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor-guide).
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] Your local test passed ./mvnw clean install -Dmaven.javadoc.skip=true.

### CODE REVIEW

**Comment by 478320** [comment]
https://github.com/apache/shenyu/issues/5623

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/controller/AlertReceiverController.java [hunk 1]
```diff
@@ -17,35 +17,34 @@
...
 import org.springframework.web.bind.annotation.DeleteMapping;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.PutMapping;
 import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.RequestParam;
-
-import jakarta.validation.Valid;
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/controller/AlertReceiverController.java:37]: not change the order

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/AlertDispatchServiceImpl.java [hunk 2]
```diff
@@ -156,6 +156,12 @@ private List<AlertReceiverDTO> matchReceiverByRules(final AlarmContent alert) {
                         return true;
                     }
 
+                    if (!Objects.isNull(item.getNamespaceId())) {
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/AlertDispatchServiceImpl.java:159]: use StringUtils.isNotBlank

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/AlertReceiverServiceImpl.java [hunk 3]
```diff
@@ -66,13 +66,13 @@ public void addReceiver(final AlertReceiverDTO alertReceiverDTO) {
         alertDispatchService.clearCache();
         alertReceiverMapper.insert(receiverDO);
     }
-    
+
     @Override
-    public void deleteReceiver(final List<String> ids) {
+    public void deleteByNamespaceId(final List<String> ids, final String namespaceId) {
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/AlertReceiverServiceImpl.java:71]: rename the method deleteReceiverByNamespaceId

**Review by moremind — CHANGES_REQUESTED** [review]

**Comment by moremind** [comment]
pls modify these changes, and check the alert module for shenyu-bootstrap

File: shenyu-plugin/shenyu-plugin-base/src/main/java/org/apache/shenyu/plugin/base/alert/AlarmSender.java [hunk 4]
```diff
@@ -75,61 +77,122 @@ public static void alarm(final AlarmContent alarmContent) {
      * @param labels Alarm labels
      */
     public static void alarm(final byte level, final String title, final String content, final Map<String, String> labels) {
+        alarm(level, title, content, labels, SYS_DEFAULT_NAMESPACE_ID);
```
Comment (xcsnx) [inline shenyu-plugin/shenyu-plugin-base/src/main/java/org/apache/shenyu/plugin/base/alert/AlarmSender.java:80]: This should not be a fixed ‘SYS_DEFAULT_NAMESPACE_ID’

**Review by xcsnx — CHANGES_REQUESTED** [review]

**Comment by moremind** [comment]
pls add my wechat: yijia1949

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/controller/AlertReceiverController.java [hunk 5]
```diff
@@ -76,37 +76,46 @@ public ShenyuAdminResult editReceiver(@Valid @RequestBody final AlertReceiverDTO
...
      * @param id alertReceiver ID
+     * @param namespaceId namespaceId.
      * @return result
      */
-    @GetMapping("/{id}")
-    public ShenyuAdminResult getReceiverDetail(@PathVariable("id") final String id) {
-        AlertReceiverDTO receiverDTO = alertReceiverService.detail(id);
+    @GetMapping("/{id}/{namespaceId}")
+    public ShenyuAdminResult getReceiverDetail(@PathVariable("id") final String id,
+                                               @PathVariable("namespaceId") @Valid
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/controller/AlertReceiverController.java:97]: the blank is error

**Review by moremind — CHANGES_REQUESTED** [review]

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/controller/AlertReceiverController.java (hunk 5 above)
Comment (moremind) [reply shenyu-admin/src/main/java/org/apache/shenyu/admin/controller/AlertReceiverController.java:97]: just use id

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/mapper/AlertReceiverMapper.java [hunk 6]
```diff
@@ -53,9 +53,10 @@ public interface AlertReceiverMapper extends ExistProvider {
      * select by primary key.
      *
      * @param id primary key
+     * @param namespaceId namespaceId
      * @return object by primary key
      */
-    AlertReceiverDO selectByPrimaryKey(String id);
+    AlertReceiverDO selectByPrimaryKey(String id, String namespaceId);
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/mapper/AlertReceiverMapper.java:59]: remove namespace

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/AlertReceiverService.java [hunk 7]
```diff
@@ -36,11 +36,12 @@ public interface AlertReceiverService {
...
 
     /**
-     * Delete alert receiver.
+     * Delete alert receiver by ids and namespaceId.
      *
      * @param ids ids
+     * @param namespaceId namespaceId
      */
-    void deleteReceiver(List<String> ids);
+    void deleteReceiverByNamespaceId(List<String> ids, String namespaceId);
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/service/AlertReceiverService.java:44]: remove namespace

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/AlertReceiverService.java [hunk 8]
```diff
@@ -66,9 +67,10 @@ public interface AlertReceiverService {
     /**
      * receiver detail.
      * @param id id
+     * @param namespaceId namespaceId
      * @return {@link AlertReceiverDTO}
      */
-    AlertReceiverDTO detail(String id);
+    AlertReceiverDTO detail(String id, String namespaceId);
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/service/AlertReceiverService.java:73]: remove namespaceId

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/AlertReceiverServiceImpl.java [hunk 9]
```diff
@@ -66,13 +66,13 @@ public void addReceiver(final AlertReceiverDTO alertReceiverDTO) {
         alertDispatchService.clearCache();
         alertReceiverMapper.insert(receiverDO);
     }
-    
+
     @Override
-    public void deleteReceiver(final List<String> ids) {
+    public void deleteReceiverByNamespaceId(final List<String> ids, final String namespaceId) {
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/AlertReceiverServiceImpl.java:71]: remove namespaceId

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/AlertReceiverServiceImpl.java [hunk 10]
```diff
@@ -96,8 +96,8 @@ public CommonPager<AlertReceiverDTO> listByPage(final AlertReceiverQuery receive
     }
     
     @Override
-    public AlertReceiverDTO detail(final String id) {
-        AlertReceiverDO receiverDO = alertReceiverMapper.selectByPrimaryKey(id);
+    public AlertReceiverDTO detail(final String id, final String namespaceId) {
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/AlertReceiverServiceImpl.java:99]: remove namspaceId

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/AlertReceiverServiceImpl.java [hunk 11]
```diff
@@ -96,8 +96,8 @@ public CommonPager<AlertReceiverDTO> listByPage(final AlertReceiverQuery receive
...
+        AlertReceiverDO receiverDO = alertReceiverMapper.selectByPrimaryKey(id, namespaceId);
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/AlertReceiverServiceImpl.java:100]: same

File: shenyu-plugin/shenyu-plugin-base/src/main/java/org/apache/shenyu/plugin/base/alert/AlarmSender.java [hunk 12]
```diff
@@ -51,12 +52,19 @@ public class AlarmSender {
...
     private static Boolean enabled;
+
+    private static String namespaceId;
     
     /**
      * Send alarm content.
      * @param alarmContent alarm content
      */
     public static void alarm(final AlarmContent alarmContent) {
+        if (alarmContent.getNamespaceId() == null) {
```
Comment (moremind) [inline shenyu-plugin/shenyu-plugin-base/src/main/java/org/apache/shenyu/plugin/base/alert/AlarmSender.java:63]: StringUtils.isNotEmpty

**Review by moremind — CHANGES_REQUESTED** [review]

**Review by moremind — APPROVED** [review]

**Review by xcsnx — APPROVED** [review]

**Pull request merged**